### PR TITLE
feat: auto-publish infoview

### PR DESCRIPTION
**Pull request by joneugster**
Automatically publish infoview and infoview-api on npmjs if the version changed.

### CODE REVIEW

**Comment by joneugster** [comment]
This is a first draft, containing the modifications to publish the infoview from CI.

One needs to create an NPM access token and add it as github secret `NPM_TOKEN`.

This does not yet contain any version number management.

I have a hard time testing this, only did so for `@leanprover/infoview` but not for the `infoview-api` yet.

**Comment by joneugster** [comment]
@mhuisi I think this is now one of the simplest versions I could think of. on a new release, if the package version of `infoview` or `infoview-api` changed, it will publish the new version to npmjs.

Using e.g. `npm --workspace=lean4-infoview version patch` to bump the version is still required to happen manually.

It does not necessarily need to be coupled to the release schedule of the VSCode-extension, but I figured doing so ensures that only maintainers will trigger the publishing (at defined steps of the procedure) and not by accident.

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -43,6 +43,18 @@ jobs:
           npm ci
           npm run build
 
+      - name: Publish infoview-api (if new version)
+        if: ${{ startsWith(github.ref, 'refs/tags/v') && !endsWith(github.ref, '-pre') && matrix.os == 'ubuntu-latest' }}
+        continue-on-error: true
+        run: |
+          [ "$(npm view @leanprover/infoview-api version)" != "$(sed -n 's/^\s*"version":\s*"\(.*\)",\s*/\1/p' lean4-infoview-api/package.json)" ] && (npm publish --workspace=lean4-infoview-api --access=public) || (echo "infoview-api version unchanged")
```
Comment (mhuisi) [inline .github/workflows/on-push.yml:50]: Is this `sed` approach really necessary? Won't publishing the InfoView simply fail if the version is unchanged?

Comment (joneugster) [reply .github/workflows/on-push.yml:50]: It does fail yes, so this is superfluous. It only shortens the CI output on failure.

Comment (mhuisi) [reply .github/workflows/on-push.yml:50]: In that case, I would prefer a solution without the `sed` invocation :-)

Comment (joneugster) [reply .github/workflows/on-push.yml]: just removed it :)

File: lean4-infoview/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
         "watch": "rollup --config --environment NODE_ENV:development --watch",
         "build": "rollup --config --environment NODE_ENV:production",
         "watchTest": "npm run watch",
-        "test": "tsc -p test/tsconfig.json"
+        "test": "./test/run-tests.sh"
```
Comment (mhuisi) [inline lean4-infoview/package.json]: It should be possible to run the tests on Windows.

Comment (joneugster) [reply lean4-infoview/package.json]: I moved that to an independent PR

**Review by Vtec234 — APPROVED** [review]
Haven't tested it but looks good in theory!

**Pull request merged**